File: app/src/main/java/com/turastory/kotlindemo/todo/app/SkipConfirmationDialog.java
package com.turastory.kotlindemo.todo.app;

import com.annimon.stream.Optional;
import com.annimon.stream.function.Consumer;
import com.turastory.kotlindemo.stub.YesOrNoDialogFragment;

public class SkipConfirmationDialog extends YesOrNoDialogFragment {
    
    private Optional<Consumer<? super YesOrNoDialogFragment>> onClickYes = Optional.empty();
    private Optional<Consumer<? super YesOrNoDialogFragment>> onClickNo = Optional.empty();
    
    public static SkipConfirmationDialog create(Consumer<? super YesOrNoDialogFragment> onClickYes) {
        SkipConfirmationDialog dialog = new SkipConfirmationDialog();
        dialog.onClickYes = Optional.ofNullable(onClickYes);
        return dialog;
    }
    
    public static SkipConfirmationDialog create(Consumer<? super YesOrNoDialogFragment> onClickYes,
                                                Consumer<? super YesOrNoDialogFragment> onClickNo) {
        SkipConfirmationDialog dialog = new SkipConfirmationDialog();
        dialog.onClickYes = Optional.ofNullable(onClickYes);
        dialog.onClickNo = Optional.ofNullable(onClickNo);
        return dialog;
    }
}

File: app/src/main/java/com/turastory/kotlindemo/kotlin/app/SkipConfirmationDialog.kt
package com.turastory.kotlindemo.kotlin.app

import com.annimon.stream.Optional
import com.annimon.stream.function.Consumer
import com.turastory.kotlindemo.stub.YesOrNoDialogFragment

class SkipConfirmationDialog : YesOrNoDialogFragment() {

    private var onClickYes = Optional.empty<Consumer<in YesOrNoDialogFragment>>()
    private var onClickNo = Optional.empty<Consumer<in YesOrNoDialogFragment>>()

    companion object {
        @JvmOverloads
        fun create(onClickYes: Consumer<in YesOrNoDialogFragment>,
                   onClickNo: Consumer<in YesOrNoDialogFragment>? = null): SkipConfirmationDialog {
            val dialog = SkipConfirmationDialog()
            dialog.onClickYes = Optional.ofNullable(onClickYes)
            dialog.onClickNo = Optional.ofNullable(onClickNo)
            return dialog
        }
    }
}
